fhir-profile/src/test/java/com/ibm/fhir/profile/test/ConstraintGeneratorTest.java

```diff
@@ -0,0 +1,226 @@
+            .type("Patient.generalPractitioner", type("Reference", profile(), targetProfile(ProfileSupport.HL7_STRUCTURE_DEFINITION_URL_PREFIX + "Organization")))
+            .cardinality("Patient.generalPractitioner", 1, "1")
+            .build();
+        ConstraintGenerator generator = new ConstraintGenerator(profile);
+        List<Constraint> constraints = generator.generate();
+        assertEquals(constraints.size(), 1);
+//      assertEquals(constraints.get(0).expression(), "generalPractitioner.exists() and generalPractitioner.all(resolve().is(Organization))");
+        assertEquals(constraints.get(0).expression(), "generalPractitioner.count() = 1 and generalPractitioner.all(resolve().is(Organization))");
+    }
+
+    @Test
+    public static void testConstraintGenerator10() throws Exception {
+        StructureDefinition profile = new ProfileBuilder(Patient.class, "http://ibm.com/fhir/StructureDefinition/TestPatient", "1.0.0")
+            .cardinality("Patient.name", 1, "1")
+            .build();
+        ConstraintGenerator generator = new ConstraintGenerator(profile);
+        List<Constraint> constraints = generator.generate();
+        assertEquals(constraints.size(), 1);
+        assertEquals(constraints.get(0).expression(), "name.count() = 1");
+    }
+
+    @Test
+    public static void testConstraintGenerator11() throws Exception {
+        StructureDefinition profile = new ProfileBuilder(Observation.class, "http://ibm.com/fhir/StructureDefinition/TestObservation", "1.0.0")
+            .binding("Observation.code", binding(BindingStrength.REQUIRED, "http://ibm.com/fhir/ValueSet/vs-1"))
+            .build();
+        ConstraintGenerator generator = new ConstraintGenerator(profile);
+        List<Constraint> constraints = generator.generate();
+        assertEquals(constraints.size(), 1);
+//      assertEquals(constraints.get(0).expression(), "code.exists() and code.memberOf('http://ibm.com/fhir/ValueSet/vs-1', 'required')");
+        assertEquals(constraints.get(0).expression(), "code.exists() and code.all(memberOf('http://ibm.com/fhir/ValueSet/vs-1', 'required'))");
+    }
+
+    @Test
+    public static void testConstraintGenerator12() throws Exception {
+        StructureDefinition profile = new ProfileBuilder(Observation.class, "http://ibm.com/fhir/StructureDefinition/TestObservation", "1.0.0")
+            .binding("Observation.code", binding(BindingStrength.REQUIRED, "http://ibm.com/fhir/ValueSet/vs-1", "http://ibm.com/fhir/ValueSet/max-vs-1"))
+            .build();
+        ConstraintGenerator generator = new ConstraintGenerator(profile);
+        List<Constraint> constraints = generator.generate();
+        assertEquals(constraints.size(), 1);
+//      assertEquals(constraints.get(0).expression(), "code.exists() and code.memberOf('http://ibm.com/fhir/ValueSet/vs-1', 'required') and code.memberOf('http://ibm.com/fhir/ValueSet/max-vs-1', 'required')");
```

Comment:
through the tests in this class, there are commented out elements, I assume these were the previous constraints?

Reply:
Removed commented out code.